File: SquareRouteProject/SquareRouteProject/app/controllers/AdminController.js
(function () {
    angular.module('SquareRoute')
        .controller('AdminController', AdminController);

    function AdminController(busStopService,routeService,accessCodeService,userService) {
        var vm = this;
        vm.message = "Admin View";

        //METHODS
        //________AccessCode________        
        //AccessCode ADD Methods
        vm.addAccessCode = function () {

            vm.input = {
                AccessCodeValue: vm.accessCodeValue_Add,
                RouteId: vm.accessCodeRouteId_Add
            };

            accessCodeService.addAccessCode(vm.input).then(callSuccess, callFail);
        }

        //AccessCode GET Methods
        vm.getAccessCodeById = function () {

            vm.input = vm.accessCodeId_Get;

            accessCodeService.getAccessCodeById(vm.input).then(callSuccess, callFail);
        }
        vm.getAccessCodeByValue = function () {

            vm.input = vm.accessCodeValue_Get;

            accessCodeService.getAccessCodeByValue(vm.input).then(callSuccess, callFail);
        }
        vm.getAllAccessCodes = function () {            
            accessCodeService.getAllAccessCodes().then(callSuccess, callFail);
        }

        //AccessCode UPDATE Methods
        vm.updateAccessCode = function () {

            vm.input = {
                AccessCodeId: vm.accessCodeId_Update,
                AccessCodeValue: vm.accessCodeValue_Update,
                RouteId: vm.accessCodeRouteId_Update
            };

            accessCodeService.updateAccessCode(vm.input).then(callSuccess, callFail);
        }

        //AccessCode DELETE Methods
        vm.deleteAccessCodeById = function () {

            vm.input = vm.accessCodeId_Delete

            accessCodeService.deleteAccessCodeById(vm.input).then(callSuccess, callFail);
        }

        //_________BusStop__________        
        //BusStop ADD Method
        vm.addBusStop = function () {

            vm.input = {
                Location: vm.busStopLocation_Add,
                RouteId: vm.busStopRouteId_Add
            };

            busStopService.addBusStop(vm.input).then(callSuccess, callFail);
        }

        //BusStop GET Methods
        vm.getBusStopById = function () {

            vm.input = vm.busStopId_Get;

            busStopService.getBusStopById(vm.input).then(callSuccess, callFail);
        }
        vm.getBusStopsByRouteId = function () {

            vm.input = vm.busStopRouteId_Get;

            busStopService.getBusStopsByRouteId(vm.input).then(callSuccess, callFail);
        }
        vm.getAllBusStops = function () {            
            busStopService.getAllBusStops().then(callSuccess, callFail);
        }

        //BusStop UPDATE Methods
        vm.updateBusStop = function () {


            vm.input = {
                BusStopId: vm.busStopId_Update,
                Location: vm.busStopLocation_Update,
                RouteId: vm.busStopRouteId_Update
            };

            busStopService.updateBusStop(vm.input).then(callSuccess, callFail);
        }

        //BusStop DELETE Methods
        vm.deleteBusStopById = function () {

            vm.input = vm.busStopId_Delete;

            busStopService.deleteBusStopById(vm.input).then(callSuccess, callFail);
        }

        //__________Route___________
        //Route ADD Method
        vm.addRoute = function () {

            vm.input = {
                RouteNum: vm.routeNum_Add,
                RouteStart: vm.routeStart_Add,
                RouteEnd: vm.routeEnd_Add,
                AccessCodeId: vm.routeAccessCodeId_Add,
                DistrictId: vm.routeDistrictId_Add
            };

            routeService.addRoute(vm.input).then(callSuccess, callFail);
        }

        //Route GET Methods
        vm.getRouteById = function () {

            vm.input = vm.routeId_Get;

            routeService.getRouteById(vm.input).then(callSuccess, callFail);
        }
        vm.getRouteByRouteNum = function () {

            vm.input = vm.routeRouteNum_Get;

            routeService.getRouteByRouteNum(vm.input).then(callSuccess, callFail);
        }
        vm.getRoutesByDistrictId = function () {

            vm.input = vm.routeDistrictId_Get;

            routeService.getRoutesByDistrictId(vm.input).then(callSuccess, callFail);
        }
        vm.getAllRoutes = function () {
            routeService.getAllRoutes().then(callSuccess, callFail);
        }

        //Route UPDATE Methods
        vm.updateRoute = function () {

            vm.input = {
                RouteId: vm.routeId_Update,
                RouteNum: vm.routeNum_Update,
                RouteStart: vm.routeStart_Update,
                RouteEnd: vm.routeEnd_Update,
                AccessCodeId: vm.routeAccessCodeId_Update,
                DistrictId: vm.routeDistrictId_Update
            };

            routeService.updateRoute(vm.input).then(callSuccess, callFail);
        }

        //Route DELETE Methods
        vm.deleteRouteById = function () {

            vm.input = vm.routeId_Delete;

            routeService.deleteRouteById(vm.input).then(callSuccess, callFail);
        }

        //__________User___________
        //User Register[ADD] Method
        vm.registerUser = function () {

            vm.input = {
                Email: vm.email,
                Password: vm.createPassword,
                ConfirmPassword: vm.confirmPassword,
                FirstName: vm.firstName,
                LastName: vm.lastName,
                ImageUrl: vm.imageUrl,
                MobileDeviceId: vm.mobileDeviceId,
                Address: vm.addressLine1 + ' ' + vm.addressLine2,
                City: vm.city,
                State: vm.state,
                PostalCode: vm.postalCode
                //RoleType: 1                
            };

            userService.registerUser(vm.input).then(callSuccess, callFail);
        }

        //User GET Methods
        vm.getUsersByRoleType = function () {

            vm.input = vm.userRoleType_Get;

            userService.getUsersByRoleType(vm.input).then(callSuccess, callFail);
        }
        vm.getUserByEmail = function () {
            
            vm.input = vm.userByEmail_Get;

            userService.getUserByEmail(vm.input).then(callSuccess,callFail);
        }
        vm.getAllUsers = function () {
            
            userService.getAllUsers().then(callSuccess, callFail);
        }

        //User UPDATE Method
        vm.updateUser = function () {

            vm.input = vm.user;

            userService.updateUser(vm.input).then(callSuccess, callFail);
        }

        //User DELETE Mehod
        vm.deleteUser = function () {

            vm.input = vm.user;

            userService.deleteUser(vm.input).then(callSuccess,callFail)
        }

        //Promise return Functions
        function callSuccess(data) {
            vm.result = data;
            console.log("Success");
            console.log(data);
        }
        function callFail(data) {                        
            vm.result = data;
            console.log("Failed");
            console.log(data);
        }
            
    }
})();

File: SquareRouteProject/SquareRouteProject/Controllers/api/AdminController.cs
using SquareRouteProject.Domain;
using SquareRouteProject.Domain.Entities;
using SquareRouteProject.Infastructure;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Http;

namespace SquareRouteProject.Presentation.Controllers.api
{
    [RoutePrefix("api/Admin")]
    public class AdminController : ApiController
    {
        private UnitOfWorkDataModels _unitOfWork;

        public AdminController()
        {
            _unitOfWork = new UnitOfWorkDataModels();
        }

        #region BusStop Methods
        #region BusStop CREATE        
        // POST api/Admin/AddBusStop
        [Route("AddBusStop")]
        public IHttpActionResult AddBusStop(BusStop busStop)
        {
            if (ModelState.IsValid) 
            {
                _unitOfWork.BusStopRepository.Repo.Add(busStop);
                _unitOfWork.SaveChanges();
            }
            return Ok();
        }
        #endregion

        #region BusStop GET        
        // GET api/Admin/GetBusStopById
        [Route("GetBusStopById")]
        public BusStop GetBusStopById(int id) 
        {
            return _unitOfWork.BusStopRepository.GetStopById(id);
        }

        // GET api/Admin/GetBusStopsbyRouteId
        [Route("GetBusStopsbyRouteId")]
        public IList<BusStop> GetBusStopsbyRouteId(int id) 
        {
            return _unitOfWork.BusStopRepository.GetStopsByRouteId(id);
        }

        // GET api/Admin/GetAllBusStops
        [Route("GetAllBusStops")]
        public IList<BusStop> GetAllBusStops() 
        {
            return _unitOfWork.BusStopRepository.Repo.GetAll();
        }
        #endregion

        #region BusStop UPDATE
        // POST api/Admin/UpdateBusStop
        [Route("UpdateBusStop")]
        public IHttpActionResult UpdateBusStop(BusStop stop) 
        {
            if (ModelState.IsValid) 
            {
                _unitOfWork.BusStopRepository.Repo.Update(stop);
                _unitOfWork.SaveChanges();
                return Ok();
            }
            return BadRequest();
        }
        #endregion

        #region BusStop DELETE
        // POST api/Admin/DeleteBusStopById
        [Route("DeleteBusStopById")]
        public IHttpActionResult DeleteBusStopById(int busStopId)
        {
            _unitOfWork.BusStopRepository.DeleteBusStopById(busStopId);
            _unitOfWork.SaveChanges();
            return Ok();
        }
        #endregion
        #endregion

        #region Route Methods
        #region Route CREATE
        // POST api/Admin/AddRoute
        [Route("AddRoute")]
        public IHttpActionResult AddRoute(Route route)
        {
            if (ModelState.IsValid)
            {
                _unitOfWork.RouteRepository.Repo.Add(route);
                _unitOfWork.SaveChanges();
                return Ok();
            }
            return BadRequest();
        }
        #endregion

        #region Route GET
        // GET api/Admin/GetRouteById
        [Route("GetRouteById")]
        public Route GetRouteById(int id)
        {
            return _unitOfWork.RouteRepository.GetRouteById(id);
        }

        // GET api/Admin/GetRouteByRouteNum
        [Route("GetRouteByRouteNum")]
        public Route GetRouteByRouteNum(int routeNum) 
        {
            return _unitOfWork.RouteRepository.GetRouteByRouteNum(routeNum);
        }

        // GET api/Admin/GetRoutesByDistrictId
        [Route("GetRoutesByDistrictId")]
        public IList<Route> GetRoutesByDistrictId(int districtId)
        {
            return _unitOfWork.RouteRepository.GetRoutesByDistrictId(districtId);
        }

        // GET api/Admin/GetAllRoutes
        [Route("GetAllRoutes")]
        public IList<Route> GetAllRoutes() 
        {
            return _unitOfWork.RouteRepository.Repo.GetAll();
        }
        #endregion

        #region Route UPDATE
        // POST api/Admin/UpdateRoute
        [Route("UpdateRoute")]
        public IHttpActionResult UpdateRoute(Route route)
        {
            if (ModelState.IsValid)
            {
                _unitOfWork.RouteRepository.Repo.Update(route);
                _unitOfWork.SaveChanges();
                return Ok();
            }
            return BadRequest();
        }
        #endregion

        #region DELETE
        // POST api/Admin/DeleteRouteById
        [Route("DeleteRouteById")]
        public IHttpActionResult DeleteRouteById(int routeId)
        {
            _unitOfWork.RouteRepository.DeleteRouteById(routeId);
            _unitOfWork.SaveChanges();
            return Ok();
        }
        #endregion
        #endregion

        #region AccessCode Methods
        #region AccessCode CREATE
        // POST api/Admin/AddAccessCode
        [Route("AddAccessCode")]
        public IHttpActionResult AddAccessCode(AccessCode accessCode) 
        {
            if (ModelState.IsValid)
            {
                _unitOfWork.AccessCodeRepository.Repo.Add(accessCode);
                _unitOfWork.SaveChanges();
                return Ok();
            }
            return BadRequest();
        }
        #endregion

        #region AccessCode GET
        // GET api/Admin/GetAccessCodeById
        [Route("GetAccessCodeById")]
        public AccessCode GetAccessCodeById(int id)
        {
            return _unitOfWork.AccessCodeRepository.GetAccessCodeById(id);
        }

        // GET api/Admin/GetAccessCodeByName
        [Route("GetAccessCodeByName")]
        public AccessCode GetAccessCodeByName(string accessCodeName)
        {
            return _unitOfWork.AccessCodeRepository.GetAccessCodeByValue(accessCodeName);
        }

        // GET api/Admin/GetAccessCodeByRouteId
        [Route("GetAccessCodeByRouteId")]
        public AccessCode GetAccessCodeByRouteId(int routeId)
        {
            return _unitOfWork.AccessCodeRepository.GetAccessCodeByRouteId(routeId);
        }

        // GET api/Admin/GetAllAccessCodes
        [Route("GetAllAccessCodes")]
        public IList<AccessCode> GetAllAccessCodes()
        {
            return _unitOfWork.AccessCodeRepository.Repo.GetAll();
        }


        #endregion
        
        #region AccessCode UPDATE
        // POST api/Admin/UpdateAccessCode
        [Route("UpdateAccessCode")]
        public IHttpActionResult UpdateAccessCode(AccessCode accessCode)
        {
            if (ModelState.IsValid)
            {
                _unitOfWork.AccessCodeRepository.Repo.Update(accessCode);
                _unitOfWork.SaveChanges();
                return Ok();
            }
            return BadRequest();
        }

        #endregion
        
        #region AccessCode Delete
        // POST api/Admin/DeleteAccessCodeById
        [Route("DeleteAccessCodeById")]
        public IHttpActionResult DeleteAccessCodeById(int accessCodeId)
        {
            _unitOfWork.AccessCodeRepository.DeleteAccessCodeById(accessCodeId);
            _unitOfWork.SaveChanges();
            return Ok();
        }        
        #endregion
        #endregion
    }
}